assets/src/edit-story/app/story/useStoryReducer/actions.js

```diff
@@ -0,0 +1,119 @@
+
+const updatePageProperties = ( dispatch ) => ( { pageIndex, properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageIndex, properties } } );
+
+const updateCurrentPageProperties = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageIndex: null, properties } } );
+
+const duplicatePage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.DUPLICATE_PAGE, payload: { pageIndex } } );
+
+const duplicateCurrentPage = ( dispatch ) => () =>
+	dispatch( { type: types.DUPLICATE_PAGE, payload: { pageIndex: null } } );
+
+const movePage = ( dispatch ) => ( { pageIndex, position } ) =>
+	dispatch( { type: types.MOVE_PAGE, payload: { pageIndex, position } } );
+
+const setCurrentPage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.SET_CURRENT_PAGE, payload: { pageIndex } } );
+
+const addElements = ( dispatch ) => ( { elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex: null, elements } } );
+
+const addElement = ( dispatch ) => ( { element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex: null, elements: [ element ] } } );
+
+const addElementsToPage = ( dispatch ) => ( { pageIndex, elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex, elements } } );
+
+const addElementToPage = ( dispatch ) => ( { pageIndex, element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex, elements: [ element ] } } );
+
+const deleteElementsById = ( dispatch ) => ( { elementIds } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds } } );
+
+const deleteSelectedElements = ( dispatch ) => () =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds: null } } );
+
+const deleteElementById = ( dispatch ) => ( { elementId } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds: [ elementId ] } } );
+
+const duplicateElementsById = ( dispatch ) => ( { elementIds } ) =>
+	dispatch( { type: types.DUPLICATE_ELEMENTS, payload: { elementIds } } );
+
+const updateElementsById = ( dispatch ) => ( { elementIds, properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds, properties } } );
+
+const updateElementById = ( dispatch ) => ( { elementId, properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds: [ elementId ], properties } } );
+
+const updateSelectedElements = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds: null, properties } } );
+
+const moveElement = ( dispatch ) => ( { elementId, position } ) =>
+	dispatch( { type: types.MOVE_ELEMENT, payload: { pageIndex: null, elementId, position } } );
+
+const moveSelectedElement = ( dispatch ) => ( { position } ) =>
+	dispatch( { type: types.MOVE_ELEMENT, payload: { pageIndex: null, elementId: null, position } } );
+
+const moveElementOnPage = ( dispatch ) => ( { pageIndex, elementId, position } ) =>
```

Comment:
Move _to_ page?

Reply:
Nope, it's if you want to reorder an element on a specific page which is *not* the current page. I don't think it'll ever be used, but is included for consistency (as the underlying operation allows it).

There's a bunch of element operations that allow manipulations to elements on other pages than the current page. Maybe all of that won't ever be needed and we can delete all of that, but I wasn't sure. I'd rather have the underlying API support it, but the exposed API limiting it if necessary.

Reply:
I see. So you are basically not going to do anything if the `pageIndex` page does not have `elementId`. Right?

Reply:
[Yes](https://github.com/ampproject/amp-wp/pull/3910/files#diff-08b4230faff7f35682a0f9f8b71366e7R12). Well, it's change to `arrangeElement` and takes a `pageId` now, but [ignores an unknown element id](https://github.com/ampproject/amp-wp/pull/3910/files#diff-08b4230faff7f35682a0f9f8b71366e7R12).